core/api/src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -35,6 +37,7 @@ const addInvoiceForSelf = async ({
       recipientWalletDescriptor,
     }: BuildWIBWithAmountFnArgs) =>
       walletInvoiceBuilder
+        .withMetadata(externalId)
```

Comment:
this should be `withExternalId`

Reply:
Yup I was wondering about whether to "future-prep" this step to take both the external id and a json blob if we decided to add that later. But having external id be its own step makes sense.

Reply:
Yeah I had the same thought - we can use 1 fn for both `{externalId, metadataJson}` that's fine. I also don't mind the class name for that state being `WithMetadata` but until we add the Json blob I would still rename the fn.

Reply:
Changed in [`force-push`](https://github.com/GaloyMoney/galoy/compare/707494475338d9ad0ebd6ebdeb47d2d16b830b1d..bce2c352786c18b90164c1b3c2519c25c489639e)